File: task14/main.cpp
#include <iostream>
#include <vector>
#include <string>
// Вариант с файлом:
// #include <fstream>
using namespace std;

vector<int> naiveSearch(string& text, string& pattern) {
    int n = text.size();
    int m = pattern.size();
    vector<int> results;

    for (int i = 0; i <= n - m; i++) {
        bool match = true;
        for (int j = 0; j < m; j++) {
            if (text[i + j] != pattern[j]) {
                match = false;
                break;
            }
        }
        if (match)
            results.push_back(i);
    }

    return results;
}

int main() {
    // Вариант с файлом:
    // ifstream f("input.txt");
    // string text, pattern;
    // getline(f, text);
    // getline(f, pattern);

    string text, pattern;
    cout << "Введите текст: ";
    getline(cin, text);
    cout << "Введите образец: ";
    getline(cin, pattern);

    vector<int> positions = naiveSearch(text, pattern);

    cout << "Вхождения на позициях: ";
    for (int pos : positions)
        cout << pos << " ";

    return 0;
}

File: task14/main.py
# Вариант с файлом:
# with open("input.txt") as f:
#     lines = f.read().splitlines()
#     text = lines[0]
#     pattern = lines[1]

def naive_search(text, pattern):
    n = len(text)
    m = len(pattern)
    results = []

    for i in range(n - m + 1):
        match = True
        for j in range(m):
            if text[i + j] != pattern[j]:
                match = False
                break
        if match:
            results.append(i)

    return results

text = input("Введите текст: ")
pattern = input("Введите образец: ")
positions = naive_search(text, pattern)
print("Вхождения на позициях:", positions)